src/d3d12/d3d12_resource_pool_texture.cpp

```diff
@@ -33,10 +34,30 @@ namespace wr
 		m_mipmapping_cpu_handle = d3d12::GetCPUHandle(m_mipmapping_heap, 0);
 		m_mipmapping_gpu_handle = d3d12::GetGPUHandle(m_mipmapping_heap, 0);
 
+		m_thread_pool = m_render_system.GetThreadPool();
+
+		m_command_lists.resize(d3d12::settings::max_threads + 1);
```

Comment:
Why max thread + 1? max_threads should never be 0 since that would mean you don't have a main thread.

Reply:
I'm using a fifth command list for mipmap generation. Combining the mipmap generation into each of the command lists breaks stuff.